File: app/src/main/java/com/example/networkaplication/persistance/repository/database/dao/DetailsDao.java
package com.example.networkaplication.persistance.repository.database.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;
import android.arch.persistence.room.Update;

import com.example.networkaplication.persistance.model.Details;

@Dao
public abstract class DetailsDao {
    @Insert(onConflict = OnConflictStrategy.ROLLBACK)
    public abstract void insert (Details details);

    @Delete
    public abstract void delete(Details details);

    @Update(onConflict = OnConflictStrategy.ROLLBACK)
    public abstract void update(Details details);

    @Query("DELETE FROM details")
    public abstract void deleteAll();

    @Query("SELECT * FROM details WHERE id=:id")
    public abstract Details findById(String id);

    @Query("SELECT * FROM details WHERE title=:title")
    public abstract Details findByTitle(String title);
}

File: app/src/main/java/com/example/networkaplication/persistance/repository/database/dao/DetailsDao.kt
package com.example.networkaplication.persistance.repository.database.dao


import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update

import com.example.networkaplication.persistance.model.Details

@Dao
abstract class DetailsDao {
    @Insert(onConflict = OnConflictStrategy.ROLLBACK)
    abstract fun insert(details: Details)

    @Delete
    abstract fun delete(details: Details)

    @Update(onConflict = OnConflictStrategy.ROLLBACK)
    abstract fun update(details: Details)

    @Query("DELETE FROM details")
    abstract fun deleteAll()

    @Query("SELECT * FROM details WHERE id=:id")
    abstract fun findById(id: String): Details

    @Query("SELECT * FROM details WHERE title=:title")
    abstract fun findByTitle(title: String): Details
}
